### PR TITLE
fix: remove duplicate permissions in rbac CLI output

### PR DESCRIPTION
**Pull request by amandavialva01**
## Description

When a user is assigned to roles that have overlapping permissions, `det rbac my-permissions` lists each permission within a given scope as many times as the number of roles assigned to the user within that scope containing that permission. 
This leads to a lot of duplicate permissions being listed within a given scope when a user runs `det rbac my-permissions`. We fix that here!

It's also worth noting that this change does not affect the result of running `det rbac my-permissions --json`.

## Test Plan

CI passes.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

DET-10172

### CODE REVIEW

File: harness/determined/cli/rbac.py [hunk 1]
```diff
@@ -89,9 +89,13 @@ def my_permissions(args: Namespace) -> None:
             workspace_name = bindings.get_GetWorkspace(sess, id=wid).workspace.name
             print(f"permissions assigned over workspace '{workspace_name}' with ID '{wid}'")
 
-        render.render_objects(
-            v1PermissionHeaders, [render.unmarshal(v1PermissionHeaders, p.to_json()) for p in perms]
-        )
+        perms_to_render = []
+        perms_added = set()
```
Comment (hamidzr) [inline harness/determined/cli/rbac.py:93]: nit: typehints

**Review by hamidzr — APPROVED** [review]
looks good I think rbac is owned by the cluster management team so I tagged them as well.

File: harness/determined/cli/rbac.py [hunk 2]
```diff
@@ -89,9 +89,13 @@ def my_permissions(args: Namespace) -> None:
...
+        for p in perms:
+            if p.id not in perms_added:
+                perms_added.add(p.id)
+                perms_to_render.append(render.unmarshal(v1PermissionHeaders, p.to_json()))
+        render.render_objects(v1PermissionHeaders, perms_to_render)
```
Comment (eecsliu) [inline harness/determined/cli/rbac.py:98]: nit: instead of building a list all over again, you can just use `list(perms_added)`

**Review by eecsliu — APPROVED** [review]
looks good! Minor nit

File: harness/determined/cli/rbac.py (hunk 2 above)
Comment (amandavialva01) [reply harness/determined/cli/rbac.py:98]: `perms_added` holds the permission ids while `perms_to_render` holds the full permissions converted to json (and ready for render). 

I tried just storing a set of json-converted permissions to have one less iterable
```
 perms_json: set[Dict [str, Any]] = set()
        for p in perms:
            p_json = p.to_json()
            if p_json not in perms_json:
                perms_json.add(p_json)
        render.render_objects(v1PermissionHeaders, 
                              [render.unmarshal(v1PermissionHeaders, p) for p in perms_json])
        print()
```
but `dicts` are unhashable so this doesn't work

**Review by azhou-determined — APPROVED** [review]
lgtm

**Pull request merged**